### PR TITLE
Add the flask extension

### PR DESCRIPTION
**Pull request by weiiwang01**
This pull request introduces an experimental Flask extension for the Juju 12-factor charm ecosystem in charmcraft.

### CODE REVIEW

**Comment by weiiwang01** [comment]
@tigarmo here's the draft pull request with the current charmcraft extension framework.

**Review by tigarmo — APPROVED** [review]

**Comment by tigarmo** [comment]
Is it possible to add a spread test for this?

**Comment by weiiwang01** [comment]
> Is it possible to add a spread test for this?

Unfortunately

File: charmcraft/extensions/flask.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                    "type": "oci-image",
+                    "description": "Prometheus exporter for statsd data",
+                    "upstream-source": "prom/statsd-exporter:v0.24.0",
+                },
+            },
+            "peers": {"secret-storage": {"interface": "secret-storage"}},
+        }
+        merging_fields = {
+            "actions": ACTIONS,
+            "options": OPTIONS,
```
Comment (syu-w) [inline charmcraft/extensions/flask.py:127]: `"config": {"options": OPTIONS},`

Comment (lengau) [reply charmcraft/extensions/flask.py:127]:
```suggestion
            "config": {"options": OPTIONS},
```

**Comment by weiiwang01** [comment]
> Is it possible to add a spread test for this?

Unfortunately, no, as we need to modify and release the flask-k8s charm as a Python library for this extension to work.

**Comment by weiiwang01** [comment]
@tigarmo @syu-w I have updated the pull request to include all the changes demonstrated in the sprint demo. Could you please review it?

**Review by syu-w — APPROVED** [review]

**Comment by syu-w** [comment]
Waiting for the spread test results.

**Pull request merged**